src/views/HardCount.vue

```diff
@@ -0,0 +1,323 @@
+import { calendarNumberOutline, checkmarkDoneOutline, storefrontOutline } from "ionicons/icons";
+import { IonBackButton, IonButton, IonCheckbox, IonContent, IonDatetime, IonFab, IonFabButton, IonHeader, IonIcon, IonInfiniteScroll, IonInfiniteScrollContent, IonInput, IonItem, IonLabel, IonList, IonListHeader, IonModal, IonPage, IonRadio, IonRadioGroup, IonSearchbar, IonSegment, IonSegmentButton, IonTitle, IonToggle, IonToolbar, onIonViewWillEnter} from "@ionic/vue";
+import { ref, nextTick, computed } from "vue"
+import { hasError, getDateTime, getDateWithOrdinalSuffix, handleDateTimeInput, showToast } from "@/utils";
+import logger from "@/logger"
+import { DateTime } from "luxon"
+import store from "@/store";
+import router from "@/router"
+import { UtilService } from "@/services/UtilService";
+
+const countName = ref(`Hard count - ${DateTime.now().toFormat('dd-MM-yyyy')}`);
+const selectedSegment = ref("group");
+const dueDate = ref("") as any;
+const queryString = ref("");
+const selectedFacilityGroupId = ref("");
+const facilities = ref([]) as any;
+const selectedFacilityIds = ref([]) as any;
+const isCountNameUpdating = ref(false);
+const dateTimeModalOpen = ref(false);
+const isAutoAssignEnabled = ref(false);
+const isScrollingEnabled = ref(false);
+const isScrollable = ref(true);
+const contentRef = ref("") as any;
+const countNameRef = ref("") as any;
+const infiniteScrollRef = ref("");
+
+const facilityGroups = computed(() => store.getters["util/getFacilityGroups"])
+
+onIonViewWillEnter(async () => {
+  await Promise.allSettled([store.dispatch("util/fetchFacilityGroups"), fetchFacilities()]);
+})
+
+async function saveCount() {
+  if(selectedSegment.value === "group" && !selectedFacilityGroupId.value) {
+    showToast(translate("Please select a facility group."))
+    return;
+  } else if(selectedSegment.value === "individual" && !selectedFacilityIds.value.length) {
+    showToast(translate("Please select atleast one facility."))
+    return;
+  }
+
+  let resp = {} as any, facilityIds = [], cycleCounts = [] as any;
+  let count = {
+    countImportName: countName.value,
+    statusId: isAutoAssignEnabled.value ? "INV_COUNT_ASSIGNED" : "INV_COUNT_CREATED",
+    countTypeEnumId: "HARD_COUNT"
+  } as any;
+
+  if(dueDate.value) count = { ...count, dueDate: dueDate.value };
+
+  if(selectedSegment.value === "group") {
+    try {
+      resp = await UtilService.fetchGroupFacilities({
+        facilityGroupId: selectedFacilityGroupId.value,
+        parentFacilityTypeId: "VIRTUAL_FACILITY",
+        parentFacilityTypeId_not: "Y",
+        facilityTypeId: "VIRTUAL_FACILITY",
+        facilityTypeId_not: "Y",
+        pageSize: 200
+      });
+
+      if(!hasError(resp)) {
+        facilityIds = resp.data.map((facility: any) => facility.facilityId);
+      } else {
+        throw resp;
+      }
+    } catch(error) {
+      logger.error(error);
+    }
+  } else {
+    facilityIds = selectedFacilityIds.value
+  }
+
+  if(!facilityIds.length) {
```

Comment:
Move this check in the catch block of fetchGroupFacilities call and update the toast message to more clear.